mantidproject/mantid
# Improve performance on event filtering

Improve performance on event filtering

**To test:**

1. Check the unit tests and doc tests.
2. Create 3 identical splitters workspace in SplittersWorkspace, MatrixWorkspace and TableWorkspace.
3. Use them to filter a same unsorted EventWorkspace an check whether the results are identical.
4. Use them with SNSPowderReduction and check whether the outputs are identical. 
5. Check the documents of FilterEvents.



Fixes #19142.

*New features are reported in docs/source/release/v3.10.0/framework.rst*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
